src/scene/mesh-instance.js

```diff
@@ -455,6 +461,32 @@ class MeshInstance {
         return bindGroup;
     }
 
+    /**
+     * Performs a raycast with a given ray against the mesh
+     *
+     * @param {Ray} ray - the ray to perform the intersect test with
+     * @returns {Vec3} Null if there is no intersection or the point of intersection
+     * in world space
+     */
+    rayCast(ray) {
+        _worldTransformInverted.copy(this.node.getWorldTransform());
+        _worldTransformInverted.invert();
+        _transformedRay.set(ray.origin, ray.direction);
+        _transformedRay.transform(_worldTransformInverted);
+        _transformedRay.direction.normalize();
+        const dist = this._mesh.rayCast(_transformedRay);
+        if (dist != null) {
+            _direction.copy(_transformedRay.direction);
+            _direction.normalize();
```

Comment:
_transformedRay.direction is normalized few lines up, no need to do this again?

Reply:
Removed _direction variable as it was redundant:

```js
if (dist) {
            _transformedRay.direction.mulScalar(dist);
            _transformedRay.origin.add(_transformedRay.direction);
            this.node.getWorldTransform().transformPoint(_transformedRay.origin, _transformedRay.origin);
            return _transformedRay.origin;
}
```